src/main/java/net/barrage/school/java/ecatalog/app/XmlPS.java

```diff
@@ -0,0 +1,73 @@
+package net.barrage.school.java.ecatalog.app;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import net.barrage.school.java.ecatalog.config.ProductSourceProperties;
+import net.barrage.school.java.ecatalog.model.Product;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+@RequiredArgsConstructor
+public class XmlPS implements ProductSource {
```

Comment:
Java-Style doesn't like cuttings like this `PS`. It's not a rule or something strict, but it's better to to call classes like this. It will be hard to read. `XmlProductSource` is always better.